optashift-employee-rostering-gwtui/src/main/java/org/optaplanner/openshift/employeerostering/gwtui/client/common/SkillConvertor.java

```diff
@@ -0,0 +1,51 @@
+package org.optaplanner.openshift.employeerostering.gwtui.client.common;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.jboss.errai.databinding.client.api.Converter;
+import org.optaplanner.openshift.employeerostering.shared.skill.Skill;
+
+public class SkillConvertor implements Converter<Set<Skill>, String> {
+
+    Map<String, Skill> skillNameToSkillMap = new HashMap<>();
+
+    @Override
+    public Class<Set<Skill>> getModelType() {
+        return (Class<Set<Skill>>) Collections.<Skill> emptySet().getClass();
+    }
+
+    @Override
+    public Class<String> getComponentType() {
+        return String.class;
+    }
+
+    public void setSkillSet(Collection<Skill> skillSet) {
+        skillNameToSkillMap.clear();
+        for (Skill skill : skillSet) {
+            skillNameToSkillMap.put(skill.getName(), skill);
+        }
+    }
+
+    @Override
+    public Set<Skill> toModelValue(String componentValue) {
+        String[] skillNames = componentValue.split(",");
+        Set<Skill> output = new HashSet<>();
+        for (String skillName : skillNames) {
+            if (skillNameToSkillMap.containsKey(skillName)) {
```

Comment:
I think we can get rid of this logic entirely (see my comment on the SingleValueTagsInputExtensions), but if we do keep it we should throw an exception when converting a skill that isn't in this map.